<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PhysicsMainWidget</class>
 <widget class="QWidget" name="PhysicsMainWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1668</width>
    <height>715</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0" colspan="2">
    <widget class="QFrame" name="frame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="1">
       <widget class="QPushButton" name="pushButton_2">
        <property name="text">
         <string>AUTOPILOT
(press any js key to stop)</string>
        </property>
        <property name="icon">
         <iconset resource="../../resources/main.qrc">
          <normaloff>:/new/prefix1/robot.png</normaloff>:/new/prefix1/robot.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QPushButton" name="aboutButton">
        <property name="toolTip">
         <string>About</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../resources/main.qrc">
          <normaloff>:/new/prefix1/info_rhombus.png</normaloff>:/new/prefix1/info_rhombus.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="pushButton_4">
        <property name="toolTip">
         <string>Camera</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../resources/main.qrc">
          <normaloff>:/new/prefix1/webcam.png</normaloff>:/new/prefix1/webcam.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="connetToVirtualButton">
        <property name="text">
         <string>connect to virtual</string>
        </property>
        <property name="icon">
         <iconset resource="../../resources/main.qrc">
          <normaloff>:/new/prefix1/server_connect.png</normaloff>:/new/prefix1/server_connect.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="pushButton_3">
        <property name="text">
         <string>connect to real device</string>
        </property>
        <property name="icon">
         <iconset resource="../../resources/main.qrc">
          <normaloff>:/new/prefix1/connect.png</normaloff>:/new/prefix1/connect.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="JoyButton">
        <property name="text">
         <string>JoyStick mode</string>
        </property>
        <property name="icon">
         <iconset resource="../../resources/main.qrc">
          <normaloff>:/new/prefix1/controller.png</normaloff>:/new/prefix1/controller.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QPushButton" name="pushButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Settings</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../resources/main.qrc">
          <normaloff>:/new/prefix1/joystick_add.png</normaloff>:/new/prefix1/joystick_add.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QFrame" name="frame_2">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>200</width>
       <height>200</height>
      </size>
     </property>
     <property name="sizeIncrement">
      <size>
       <width>500</width>
       <height>0</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="maximumSize">
         <size>
          <width>350</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Contols</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QComboBox" name="comboBox">
           <property name="currentText">
            <string/>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="CopterInputsWidget" name="inputsWidget" native="true"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>3D View</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="CloudViewDialog" name="cloud" native="true"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Camera</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="0">
           <widget class="AdvancedImageWidget" name="imageView" native="true"/>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="LoggerWidget" name="logWidget" native="true">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>100</height>
      </size>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CloudViewDialog</class>
   <extends>QWidget</extends>
   <header location="global">cloudViewDialog.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>AdvancedImageWidget</class>
   <extends>QWidget</extends>
   <header location="global">advancedImageWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CopterInputsWidget</class>
   <extends>QWidget</extends>
   <header>copterInputsWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LoggerWidget</class>
   <extends>QWidget</extends>
   <header>loggerWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../resources/main.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>PhysicsMainWidget</receiver>
   <slot>startRealMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>182</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>503</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>JoyButton</sender>
   <signal>clicked()</signal>
   <receiver>PhysicsMainWidget</receiver>
   <slot>startJoyStickMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>373</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>229</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>connetToVirtualButton</sender>
   <signal>released()</signal>
   <receiver>PhysicsMainWidget</receiver>
   <slot>startVirtualMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>522</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>833</x>
     <y>357</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>released()</signal>
   <receiver>PhysicsMainWidget</receiver>
   <slot>settingsWidget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1637</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>1662</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>aboutButton</sender>
   <signal>released()</signal>
   <receiver>PhysicsMainWidget</receiver>
   <slot>aboutWidget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1653</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>1661</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>released()</signal>
   <receiver>PhysicsMainWidget</receiver>
   <slot>startCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1631</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>1586</x>
     <y>167</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>myHandler(int)</slot>
  <slot>rollChange(int)</slot>
  <slot>yawChange(int)</slot>
  <slot>throttleChange(int)</slot>
  <slot>pitchChange(int)</slot>
  <slot>startRealMode()</slot>
  <slot>startJoyStickMode()</slot>
  <slot>CH5Change(int)</slot>
  <slot>CH6Change(int)</slot>
  <slot>CH7Change(int)</slot>
  <slot>CH8Change(int)</slot>
  <slot>startVirtualMode()</slot>
  <slot>settingsWidget()</slot>
  <slot>aboutWidget()</slot>
  <slot>startCamera()</slot>
 </slots>
</ui>
